pkg/netlink/netlink.go

```diff
@@ -93,6 +93,31 @@ var (
 	}
 )
 
+const (
+	// MaxRetries is the maximum number of retries for interrupted netlink operations
+	MaxRetries = 5
+)
+
+// retryOnInterrupted wraps a function call and retries it if interrupted
+func retryOnInterrupted[T any](operation func() (T, error)) (T, error) {
+	var result T
+	var err error
+
+	for i := 0; i < MaxRetries; i++ {
+		result, err = operation()
+		if err == nil || !errors.Is(err, netlink.ErrDumpInterrupted) {
+			return result, err
+		}
+
+		if i < MaxRetries-1 {
+			logger.V(log.DEBUG).Infof("Netlink operation interrupted (attempt %d/%d), retrying: %v", i+1, MaxRetries, err)
+		}
+	}
+
+	logger.Warningf("Netlink operation failed after %d retries due to interruption: %v", MaxRetries, err)
+	return result, err
```

Comment:
We can simplify and use `retry.OnError`:
```suggestion
     backOff := wait.Backoff{
		Steps:    MaxRetries,
		Duration: 2=10 * time.Millisecond,
		Factor:   1.0,
		Jitter:   0.1,
	}

	err := retry.OnError(backOff, func(err error) bool {
		return errors.Is(err, netlink.ErrDumpInterrupted)
	}, func() error {
		var opErr error

		result, opErr = operation()
		return opErr
	})

	return result, err
```